src/com/microsoft/azure/documentdb/GatewayProxy.java

```diff
@@ -154,7 +163,13 @@ private HttpClient getHttpClient(boolean isForMedia) {
      * @return the created HttpClient
      */
     private HttpClient createHttpClient(boolean isForMedia) {
-        HttpClient httpClient = new DefaultHttpClient(this.connectionManager);
+        ProxySelectorRoutePlanner ps = new ProxySelectorRoutePlanner(
+                SchemeRegistryFactory.createDefault(), ProxySelector.getDefault());
+
+        DefaultHttpClient defaultHttpClient = new DefaultHttpClient(this.connectionManager);
+        defaultHttpClient.setRoutePlanner(ps);
+        
+        HttpClient httpClient = defaultHttpClient;
         HttpParams httpParams = httpClient.getParams();
```

Comment:
Remove this line "HttpClient httpClient = defaultHttpClient;". And change the next line to be "HttpParams httpParams = defaultHttpClient.getParams();"